### PR TITLE
Fix the build on nightly

### PR DESCRIPTION
**Pull request by steveklabnik**
and fix two warnings

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,5 +18,5 @@ api_test = []
 
 [dependencies]
 rustc-serialize = "~0.3"
-hyper = "~0.5"
+hyper = "0.6.4"
```
Comment (jwilm) [inline Cargo.toml:21]: Will this force executable dependents to use `0.6.4` specifically? Would it be better to have a more flexible semver spec like `^0.6.4`? The way I understand it, there is a set of symbols for each version of a dependency in the final product.

**Comment by jwilm** [comment]
Hi @steveklabnik. Thanks for the PR! Just had one question on the diff (above).

Do you happen to be using the library, or is this PR a community service? I ask since I noticed in my feed that you made a series of forks on a bunch of libraries. In either case, the contribution is greatly appreciated! :grin:

File: Cargo.toml (hunk 1 above)
Comment (steveklabnik) [inline Cargo.toml:21]: This is equivalent to `^0.6.4`

**Comment by steveklabnik** [comment]
Just community service :)

**Comment by jwilm** [comment]
That's super cool of you :sunglasses:. Thanks again.

**Comment by steveklabnik** [comment]
Any time. :) You might want to update the crates.io version as well, so people get the fixes

**Pull request merged**